snmp/mysql-stats

```diff
@@ -57,75 +78,113 @@ cursor = conn.cursor()
+            data['ib_bpool_free'] = ib_bpool_free.group(1)
+        elif ib_bpool_dbpages:
+            data['ib_bpool_dbpages'] = ib_bpool_dbpages.group(1)
+        elif ib_bpool_modpages:
+            data['ib_bpool_modpages'] = ib_bpool_modpages.group(1)
+        elif ib_io:
+            data['ib_io_read'] = ib_io.group(1)
+            data['ib_io_write'] = ib_io.group(2)
+            data['ib_io_fsync'] = ib_io.group(3)
+        elif ib_io_log:
+            data['ib_io_log'] = ib_io_log.group(1)
+        elif ib_io_p1:
+            data['ib_iop_aioread'] = ib_io_p1.group(1)
+            data['ib_iop_aiowrite'] = ib_io_p1.group(2)
+        elif ib_io_p2:
+            data['ib_iop_ibuf_aio'] = ib_io_p2.group(1)
+            data['ib_iop_log'] = ib_io_p2.group(2)
+            data['ib_iop_sync'] = ib_io_p2.group(3)
+        elif ib_io_p3:
+            data['ib_iop_flush_log'] = ib_io_p3.group(1)
+            data['ib_iop_flush_bpool'] = ib_io_p3.group(2)
+        elif ib_log_p1:
+            data['ib_log_written'] = ib_log_p1.group(1)
+            if ib_log_p1.group(2):
+                data['ib_log_written'] = int(data['ib_log_written']) + int(ib_log_p1.group(2))
+        elif ib_log_p2:
+            data['ib_log_flush'] = ib_log_p2.group(1)
+            if ib_log_p2.group(2):
+                data['ib_log_flush'] = int(data['ib_log_flush']) + int(ib_log_p2.group(2))
+        elif ib_semaphore:
+            data['ib_spin_waits'] = ib_semaphore.group(1)
+            data['ib_spin_rounds'] = ib_semaphore.group(2)
+            data['ib_os_waits'] = ib_semaphore.group(3)
+        elif ib_tnx:
+            data['ib_tnx'] = ib_tnx.group(1)
+            if ib_tnx.group(2):
+                data['ib_tnx'] = int(data['ib_tnx']) + int(ib_tnx.group(2))
+        elif ib_b_reg:
+            data['ib_bpool_read'] = ib_b_reg.group(1)
+            data['ib_bpool_created'] = ib_b_reg.group(2)
+            data['ib_bpool_written'] = ib_b_reg.group(3)
 
+points_data = []
+points_label = []
 for category in datavariables:
     for variable in datavariables[category]:
-	if variable in data:
-	    print data[variable]
+        if variable in data:
+            points_label.append(variable.lower())
+            points_data.append(int(float(data[variable])))
+
+entries = dict(zip(points_label, points_data))
+
+if os.path.isfile(temp_file):
+    with open(temp_file, 'r') as file:
+        old_data = json.loads(file.read().replace("\'","\""))
+
+    new_points_data = []
+    new_points_label = []
+
+    for point in old_data:
+        new_points_label.append(point)
+        new_points_data.append(int(entries[point]) - int(old_data[point]))
+
+    new_entries = dict(zip(new_points_label, new_points_data))
+else:
+    new_entries = json.loads(str(entries).replace("\'","\""))
+
+with open(temp_file, 'w') as file:
+    file.write(str(entries))
+
+export = {"data": new_entries, "error": "0", "errorString": "", "version": "1"}
+data = re.sub(' ', '', json.dumps(export))
```

Comment:
space removement not needed
Example:
https://github.com/librenms/librenms-agent/pull/258/files